txtrie/txtrie_test.go

```diff
@@ -81,25 +82,26 @@ func addTrie(txTries *TxTries, root common.Hash, transactions []common.Hash, db
 
 }
 
-func computeEthReferenceTrieHash(transactions []common.Hash) (common.Hash, error) {
+func computeEthReferenceTrieHash(transactions types.Transactions) (common.Hash, error) {
 	db := createReferenceDB()
 	newTrie, err := trie.New(emptyRoot, trie.NewDatabaseWithCache(db, 0, ""))
 	if err != nil {
 		return emptyHash, err
 	}
 
 	for i, tx := range transactions {
-		b, err := intToBytes(i)
+
+		key, err := rlp.EncodeToBytes(uint(i))
 		if err != nil {
 			return emptyHash, err
 		}
 
-		key, err := rlp.EncodeToBytes(b)
+		value, err := rlp.EncodeToBytes(tx)
 		if err != nil {
 			return emptyHash, err
 		}
 
-		newTrie.Update(key, tx.Bytes())
+		newTrie.Update(key, value)
```

Comment:
i think its better to use function newTrie.TryUpdate since it does returns error, while Update only logs it

Reply:
updated